src/runtime/service_engine.h

```diff
@@ -128,6 +128,9 @@ class service_engine : public utils::singleton<service_engine>
     service_spec _spec;
     env_provider *_env;
 
+    dsn_handle_t _get_runtime_info_cmd;
```

Comment:
I would recommend wrapping `command_instance*` into a RAII class, so that we do not need to UNREGISTER_VALID_HANDLER manually. If we don't refactor in this PR, I will take this later.